pkg/util/helper/binding_test.go

```diff
@@ -180,3 +196,695 @@ func TestObtainBindingSpecExistingClusters(t *testing.T) {
+				"apiVersion": "v1",
+				"kind":       "Node",
+				"metadata": map[string]interface{}{
+					"name":              "node",
+					"creationTimestamp": nil,
+				},
+			}},
+			wantErr: false,
+		},
+		{
+			name: "cluster scope: get from cache",
+			args: args{
+				dynamicClient: dynamicfake.NewSimpleDynamicClient(scheme.Scheme),
+				informerManager: func(stopCh <-chan struct{}) genericmanager.SingleClusterInformerManager {
+					c := dynamicfake.NewSimpleDynamicClient(scheme.Scheme,
+						&corev1.Node{ObjectMeta: metav1.ObjectMeta{Name: "node"}})
+					m := genericmanager.NewSingleClusterInformerManager(c, 0, stopCh)
+					m.Lister(corev1.SchemeGroupVersion.WithResource("nodes"))
+					m.Start()
+					m.WaitForCacheSync()
+					return m
+				},
+				restMapper: func() meta.RESTMapper {
+					m := meta.NewDefaultRESTMapper([]schema.GroupVersion{corev1.SchemeGroupVersion})
+					m.Add(corev1.SchemeGroupVersion.WithKind("Node"), meta.RESTScopeRoot)
+					return m
+				}(),
+				resource: workv1alpha2.ObjectReference{
+					APIVersion: "v1",
+					Kind:       "Node",
+					Name:       "node",
+				},
+			},
+			want: &unstructured.Unstructured{Object: map[string]interface{}{
+				"apiVersion": "v1",
+				"kind":       "Node",
+				"metadata": map[string]interface{}{
+					"name":              "node",
+					"creationTimestamp": nil,
+				},
+			}},
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			stopCh := make(chan struct{})
+			mgr := tt.args.informerManager(stopCh)
+			got, err := FetchWorkload(tt.args.dynamicClient, mgr, tt.args.restMapper, tt.args.resource)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("FetchWorkload() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if got != nil {
+				delete(got.Object, "spec")
+				delete(got.Object, "status")
+			}
+			if !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("FetchWorkload() got = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestDeleteWorkByRBNamespaceAndName(t *testing.T) {
+	type args struct {
+		c         client.Client
+		namespace string
+		name      string
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    []workv1alpha1.Work
+		wantErr bool
+	}{
+		{
+			name: "delete",
+			args: args{
+				c: fake.NewClientBuilder().WithScheme(gclient.NewSchema()).WithObjects(
+					&workv1alpha1.Work{
+						ObjectMeta: metav1.ObjectMeta{
+							Name: "w1", Namespace: names.ExecutionSpacePrefix + "cluster",
+							Labels: map[string]string{
+								workv1alpha2.ResourceBindingReferenceKey: names.GenerateBindingReferenceKey("default", "foo"),
+							},
+							Annotations: map[string]string{
+								workv1alpha2.ResourceBindingNameAnnotationKey:      "foo",
+								workv1alpha2.ResourceBindingNamespaceAnnotationKey: "default",
+							},
+						},
+					},
+				).Build(),
+				namespace: "default",
+				name:      "foo",
+			},
+			want:    []workv1alpha1.Work{},
+			wantErr: false,
+		},
```

Comment:
Can here add more cases: the deleted object not found, delete failed?

Reply:
I don't find a way to simulate delete fail with fake client~

Reply:
Okay, I see.